src/Sentry.Protocol/BaseScopeExtensions.cs

```diff
@@ -128,11 +128,6 @@ public static bool HasUser(this BaseScope scope)
         /// <param name="breadcrumb">The breadcrumb.</param>
         internal static void AddBreadcrumb(this BaseScope scope, Breadcrumb breadcrumb)
         {
-            if (scope == null)
```

Comment:
We discussed this on discord and disagreed so I believe I should try to clarify my world view here so we can at least make sure we understand each side before we agree which direction to go.

In a pure C# 8 world, this change is correct. The compiler will make sure you can't:

```csharp
BaseScope scope = null;
scope.AddBreadcrumb(..);
```

But in the context of this code base, which includes a target `net461`, this could be problematic.

The potential problem I see is that an application targeting .NET 4.6.1 will compile just fine with the snippet above. The tooling at that version is unaware of the nullable annotations added by the C# 8 compiler. Which means, that in that case it could lead to a runtime error.

Options:

1.
```csharp

#if HAS_NULLABLE_REF_TYPE (or basically NET461)
if (scope == null) return;
#endif
```

2. 

```
#pragma CODE // net461 could pass null here
if (scope == null) return;
```

3. 

A better way?

Reply:
> The potential problem I see is that an application targeting .NET 4.6.1 will compile just fine with the snippet above. The tooling at that version is unaware of the nullable annotations added by the C# 8 compiler. Which means, that in that case it could lead to a runtime error.

That's actually not true. We use `Nullable` package for polyfilling, so the users on net461 will get the same NRT experience as on any other framework. Also, it's used for the same purpose for any target framework below netstandard2.1, which is basically everything Sentry has been targeting till now.

Reply:
> The only nuance is that C#8 is default on netcore3.0+ and netstd2.1+ but this is not the case on older targets, so the user will have to see the <LangVersion> themselves to benefit from NRTs and other features (but again, this is not just net461). This can be highlighted in docs.

Sadly this is an issue though. Most of users, for a long time, will simply update the SDK and not use C# 8 or newer targets.

In that case, it could throw at runtime.